jsbits/Scenario.js

```diff
@@ -192,10 +194,22 @@ function gm$smartProcessFeatureCollection(fc, defVec, maxGeomId) {
     points.forEach(addGeomID);
     lines.forEach(addGeomID);
     surfaces.forEach(addGeomID);
+
+    // Get lat, lon, alt, srid if available
+    if (gi['lat'] && gi['lat'].constructor === Number &&
+        gi['lon'] && gi['lon'].constructor === Number &&
+        gi['alt'] && gi['alt'].constructor === Number) {
+        lat = gi['lat'];
+        lon = gi['lon'];
+        alt = gi['alt'];
+    }
+    if (gi['srid'] && gi['srid'].constructor === Number) {
+        srid = gi['srid']
+    }
+
     // transform everything from WGS84 to a metric reference system if needed
-    if(cmin.length >= 2 && cmax.length >= 2
-       && cmin[0] > -360 && cmax[0] < 360 && cmin[1] > -180 && cmax[1] < 180
-       && (cmax[0] - cmin[0]) < 5 && (cmax[1] - cmin[1]) < 5) {
```

Comment:
I think, we still should keep some heuristics to decide if this is WGS'84 or metric system.
My idea behind the step 1 was to enhance it somehow to make it more robust.